rpc/handlers.go

```diff
@@ -1120,3 +1123,49 @@ func (h *Handler) EstimateFee(broadcastedTxns []BroadcastedTransaction, id Block
 
 	return estimates, nil
 }
+
+func (h *Handler) TraceTransaction(hash felt.Felt) (TransactionTrace, *jsonrpc.Error) {
+	_, _, blockNumber, err := h.bcReader.Receipt(&hash)
+	if err != nil {
+		return nil, jsonrpc.Err(jsonrpc.InternalError, err.Error())
+	}
+	tx, err := h.bcReader.TransactionByHash(&hash)
+	if err != nil {
+		return nil, jsonrpc.Err(jsonrpc.InternalError, err.Error())
+	}
+
+	id := &BlockID{Number: blockNumber - 1}
+
+	state, closer, err := h.stateByBlockID(id)
+	if err != nil {
+		return nil, ErrBlockNotFound
+	}
+	defer func() {
+		if closeErr := closer(); closeErr != nil {
+			h.log.Warnw("Failed to close state in starknet_call", "err", closeErr)
+		}
+	}()
+
+	header, err := h.blockHeaderByID(id)
+	if err != nil {
+		return nil, ErrBlockNotFound
+	}
+
+	var classes []core.Class
+	if tx, ok := tx.(*core.DeclareTransaction); ok {
+		class, stateErr := state.Class(tx.ClassHash)
+		if stateErr != nil {
+			return nil, jsonrpc.Err(jsonrpc.InternalError, stateErr.Error())
+		}
+		classes = append(classes, class.Class)
+	}
+
+	_, info, err := h.vm.Execute([]core.Transaction{tx}, classes, header.Number, header.Timestamp, header.SequencerAddress, state, h.network)
+	if err != nil {
+		err = fmt.Errorf("vm.Execute: %w", err)
+		return nil, jsonrpc.Err(jsonrpc.InternalError, err.Error())
```

Comment:
```suggestion
                rpcErr := *ErrContractError
                rpcErr.Data = err.Error()
                return nil, &rpcErr
```

Let's do what EstimateFee does